DEVELOPMENT-GUIDE.md

```diff
@@ -0,0 +1,85 @@
+# ACCP Development Guidance
+Correctness and code-safety is paramount in the Amazon Corretto Crypto Provider.
+
+The purpose of this guide is not to provide all information needed for development within ACCP.
+Instead, it is intended to provide a quick introduction to the most important (and ACCP-specific) components to help developers find and learn what is most important.
+
+# Development Principles
+
+In decreasing order of importance:
+
+1. We must be secure.
+    Using ACCP must *never* cause an application to be less secure than if it weren't used.
+2. Correctness.
+    We must _never_ do the wrong thing or return the wrong result.
+    Subtle failures are easy to miss and can cause problems.
+    When something goes wrong we should fail obviously and force the error to be properly handled.
+3. All logic flows must be designed and explicit.
+    This means that there must be no dependencies on undefined behavior (in Java or C++).
+    (There is a single exception to this rule. We are permitted to depend on implicit `NullPointerException`s in Java.)
+4. Testing ensures correctness.
+    1. Known answer and compatibility tests are crucial.
+        As we are implementing standards, we must be compatible with other implementations. Both the default implementation from Java and BouncyCastle are considered acceptable alternatives. (They both have had historical bugs and so both are used to work around issues in the other.)
```

Comment:
Does "As" mean "because" or "while". Please clarify.

Reply:
"Because". I'll fix it.